psi4/driver/procrouting/proc.py

```diff
@@ -2291,6 +2297,17 @@ def run_scf(name, **kwargs):
     for k, v in scf_wfn.variables().items():
         core.set_variable(k, v)
 
+
+    # optionally get excited states
+    if do_tdscf:
+        excited_states = run_tdscf_energy(name, ref_wfn=scf_wfn)
+    
+        # This way of storing energies and states seems ... not ideal
+        for n, ex_pair in enumerate(excited_states): 
+            ex_en, ex_state = ex_pair
+            scf_wfn.set_variable(f'STATE {n + 1} EXCITATION ENERGY', ex_en) 
+            scf_wfn.set_variable(f'STATE {n + 1} IRREP', ex_state) 
```

Comment:
Should these be set here in `proc.py`, or does it make more sense for them to be set in the tdscf code itself? The latter would allow storage of more things in wfn (eg absolute energies of excited states) without requiring a complex return type from `tdscf_excitations`.

Reply:
I agree it makes more sense to set these on `Wavefunction` directly in the TDSCF code.

Reply:
Done, I also like Lori's suggestion for the variable keys.